telemetry/src/server.rs

```diff
@@ -21,7 +21,7 @@ impl Route {
 }
 
 pub fn run(telemetry_state: Arc<TelemetryState>, port: u16) {
-    rouille::start_server(("localhost", port), move |request| {
+    rouille::start_server(("0.0.0.0", port), move |request| {
```

Comment:
This really should be a config option someplace. Hardcoding either of these values is bad news when running in production.

Reply:
OK, thanks @mircea-c, we do have a config that you can only specify the port. 

```
[telemetry]
enabled = true
port = 3001
```
 should we add a host parameter ? or replace the port with an `url` parameter where you specify a full URL ?

Reply:
Issues caused by configuration are easy to fix. Hardcoding this to accept requests from every network interface on the machine can become a security issue. 

Telemetry should be considered a separate service from the relayer (which also shouldn't just listen to all interfaces by default) and have separate networking / access configuration. The operator needs to be allowed to decide what network interface / port the telemetry will be exposed on.

Reply:
OK, that's fair. The telemetry might run from a server with multiple NICs

do you have a preference for having just one parameter like 
```
[telemetry]
enabled = true
address = '127.0.0.1:3001'
```
or
```
[telemetry]
enabled = true
port = 3001
host = '127.0.0.1'
```

Reply:
I would prefer one parameter like `address`, but that's just me :slightly_smiling_face:
Access is fine as is, I wouldn't expect anything else.

Reply:
I would rather split the port and host into two options, as it makes clear that the port is required, otherwise it's too easy to just leave it out.

Reply:
I took the liberty of doing this in https://github.com/informalsystems/ibc-rs/pull/1033/commits/428c45a0ea805fa804495366c96c9111c61d087e.

Reply:
Lgtm! Thanks!

Reply:
@romac, testing locally, one use case is that if something is already running on the specified port (e.g. `3001`), when you start the relayer with `Hermes start` you get this message. Is this OK or should we not start it ?

```
./target/debug/hermes -c ./config.toml start
The application panicked (crashed).
Message:  Failed to start server: Os { code: 48, kind: AddrInUse, message: "Address already in use" }
Location: $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/rouille-3.1.1/src/lib.rs:221
```

Reply:
I think it's fine to crash, as otherwise one may not realize that the port was already in use and end up running Hermes without telemetry, effectively. We could perhaps show a nice error message and gracefully abort rather than panic?

Reply:
Oh and we also need a changelog entry for this.

Reply:
I'd say just showing a friendly message saying telemetry not enable because port is in use should be fine. Not showing `The application panicked` should be better

Reply:
> Oh and we also need a changelog entry for this.

Isn't telemetry added in this release? I would just keep one entry for adding the feature and maybe list all issues that created and fixed things.

Reply:
re: "Address already in use" ...let's exit gracefully. The user should then either disable telemetry or try with a different address/port.